enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -465,22 +429,43 @@ There are several drawbacks with these approaches:
 
 - `HostPath` volumes require permissions that we do not want to give to the users targeted for consumption.
 Current understanding based on a conversation between @smarterclayton, @bparees, and @gabemontero:
->> - Typically, to read anything on the `HostPath` volume, the privileged bit needs to be set
->> - Some additional support, centered around `SecurityContextContraints` and SELinux, does exist as an add-on in OpenShift
->> (vs. standard K8S), and it allows users to mount `HostPath` volumes prepared for them, if they include necessary SELinux labelling.
->> - So the cluster admin sets that SELinux label on the `HostPath` volume and namespace where it will be consumed.
->> - a new `SecurityContextConstraint` that is based off of the `hostpath-anyuid` SCC (which is defined on install) and
->> includes the SELinux label can be created.
->> - then the serviceaccount for the user Pod in question would be given access to the new SCC
->> - However, the cluster administrator has to be careful with the mount provided.
->> - Mounts at certain parts of the filesystem can basically root the node
-- Host injection via the MCO requires a node reboot for the updates to take effect.  That is generally deemed
-undesirable for this feature.
-- The MCO API has not been intuitive for the type of users to date who are interested in introducing and then consuming
-small collections of files on a host.  The RHCOS team has experienced challenges in coaching consumers who want to put
-entitlements on 4.x nodes via the MCO.
-
-### CSI specific alternatives
+  - Typically, to read anything on the `HostPath` volume, the privileged bit needs to be set
+  - Some additional support, centered around `SecurityContextContraints` and SELinux, does exist as an add-on in OpenShift (vs. standard K8S), and it allows users to mount `HostPath` volumes prepared for them if they include necessary SELinux labelling.
+  - So the cluster admin sets that SELinux label on the `HostPath` volume and namespace where it will be consumed.
+    Admin would need to create a new `SecurityContextConstraint` that is based off of the `hostpath-anyuid` SCC (which is defined on install) and includes the SELinux label can be created.
+  - However, the cluster administrator has to be careful with the mount provided.
+    Mounts at certain parts of the filesystem can basically root the node.
+- Host injection via the MCO requires a node reboot for the updates to take effect.
+  That is generally deemedundesirable for this feature.
+- The MCO API has not been intuitive for the type of users to date who are interested in introducing and then consuming small collections of files on a host.
+  The RHCOS team has experienced challenges in coaching consumers who want to put entitlements on 4.x nodes via the MCO.
+
+### Implement a provider for the Secret Store CSI Driver
```

Comment:
@jsafrane @gabemontero @deads2k @derekwaynecarr @smarterclayton 

This is an alternative approach that we should think about, especially if we want to present something upstream.
At minimum we should have an answer as to why we _don't_ want to pursue this approach.

In many respects this approach can work as long as we get the RBAC story right.
The secret store CSI driver (at present) restricts use on the namespace level.
We could propose SARs on the `SecretProvierClass` object as an opt-in feature.
The cluster-wide `Share` object then becomes a way to control which content is globally available.

We'd also want a means to configure the secret store providers via the Cluster Storage Operator.
There are some secret store providers that are cloud provider specific, as well as HashiCorp's Vault provider that maintained by a third party.

Reply:
based on our conversation this morning @adambkaplan do you think we've properly captured the disadvantages of this approach?  both in the EP text and comments here?

the entire driver/api is still alpha
secrets per namespace in addition to SecretProviderClass per namespace (IIRC)I f
I feel like I'm forgetting an item or 2 ....